frame/staking/src/pallet/mod.rs

```diff
@@ -184,8 +184,36 @@ pub mod pallet {
 		///
 		/// The changes to nominators are reported to this. Moreover, each validator's self-vote is
 		/// also reported as one independent vote.
+		///
+		/// To keep the load off the chain as much as possible, changes made to the staked amount
+		/// via rewards and slashes are not reported and thus need to be manually fixed by the
+		/// staker. In case of `bags-list`, this always means using `rebag` and `putInFrontOf`.
+		///
+		/// Invariant: what comes out of this list will always be a nominator.
```

Comment:
Can add this to `try_state`?

Reply:
Already done here: https://github.com/paritytech/substrate/blob/8f54be58e437cceb275c2ea2c75c146846fb5e35/frame/staking/src/pallet/impls.rs#L1529

Reply:
But do we check this particular invariant anywhere? I don't think so. I was expecting a check like: 

```
T::VoterList::iter().collect().all(|x| x.is_nominator())
```

Reply:
Ah, it's always confusing to me when I see a bunch of comments, now it's clear.